internal/controllers/certificaterequest_controller.go

```diff
@@ -76,23 +80,6 @@ func (r *CertificateRequestReconciler) Reconcile(ctx context.Context, req ctrl.R
 		return ctrl.Result{}, nil
 	}
 
-	if r.CheckApprovedCondition {
-		// If CertificateRequest has not been approved or is denied, exit early.
-		if !cmutil.CertificateRequestIsApproved(&certificateRequest) || cmutil.CertificateRequestIsDenied(&certificateRequest) {
-			log.Info("certificate request has not been approved")
-			return ctrl.Result{}, nil
-		}
-	}
-
-	// Ignore CertificateRequest if it is already Ready
-	if cmutil.CertificateRequestHasCondition(&certificateRequest, cmapi.CertificateRequestCondition{
-		Type:   cmapi.CertificateRequestConditionReady,
-		Status: cmmeta.ConditionTrue,
-	}) {
-		log.Info("CertificateRequest is Ready. Ignoring.")
-		return ctrl.Result{}, nil
-	}
```

Comment:
I think this ought to stay here.
If the Ready condition has been set to True, then the issuer should make no further changes to it.

And the same goes for the other terminal states: `Ready: False, Reason: cmapi.CertificateRequestReasonDenied` and `Ready: False, Reason: cmapi.CertificateRequestReasonFailed`.

This will avoid the controller repeatedly updating the `.Status.FailureTime` below, if triggered by other CR status changes or metadata changes.
(the CR.Spec is immutable now, right?)

Reply:
That makes sense to me